### PR TITLE
upgrade app (sdk+ rn + node)

### PR DESCRIPTION
**Pull request by hilat-autofleet**
upgrade

### CODE REVIEW

File: examples/client/Locomotion/.node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+16
```
Comment (hilat-autofleet) [inline examples/client/Locomotion/.node-version:1]: 16.20.2

File: examples/client/Locomotion/package.json [hunk 2]
```diff
@@ -104,33 +106,37 @@
...
     "eslint-watch": "^3.1.5",
-    "jest": "^26.6.3",
+    "jest": "^29.2.1",
     "jetifier": "^2.0.0",
-    "metro-react-native-babel-preset": "^0.67.0",
+    "metro-react-native-babel-preset": "^0.73.5",
     "patch-package": "^7.0.0",
-    "react-test-renderer": "17.0.2",
-    "typescript": "^4.6.4"
+    "prettier": "^2.4.1",
```
Comment (OmerGery) [inline examples/client/Locomotion/package.json]: we use/need prettier?

File: examples/client/Locomotion/package.json [hunk 3]
```diff
@@ -24,17 +24,18 @@
...
+    "@react-native-community/geolocation": "^3.1.0",
+    "@react-native-community/netinfo": "^11.3.2",
+    "@react-native-firebase/app": "^20.4.0",
+    "@react-native-firebase/crashlytics": "^20.4.0",
     "@react-native-picker/picker": "^2.4.1",
     "@react-navigation/drawer": "^6.4.1",
     "@react-navigation/native": "^6.0.10",
     "@react-navigation/native-stack": "^6.6.2",
-    "@stripe/stripe-react-native": "^0.14.0",
+    "@stripe/stripe-react-native": "^0.20.0",
```
Comment (OmerGery) [inline examples/client/Locomotion/package.json:38]: extra testing to stripe

File: examples/client/Locomotion/package.json [hunk 4]
```diff
@@ -24,17 +24,18 @@
...
     "@react-native-async-storage/async-storage": "^1.17.3",
-    "@react-native-community/cli": "^7.0.4",
+    "@react-native-community/cli": "^10.2.7",
+    "@react-native-community/cli-platform-android": "^10.2.0",
     "@react-native-community/clipboard": "^1.5.1",
-    "@react-native-community/geolocation": "^2.0.2",
-    "@react-native-community/netinfo": "^9.3.7",
-    "@react-native-firebase/app": "^14.11.1",
-    "@react-native-firebase/crashlytics": "^14.11.1",
+    "@react-native-community/geolocation": "^3.1.0",
```
Comment (OmerGery) [inline examples/client/Locomotion/package.json:30]: we will need testing for live location/ road test

File: examples/client/Locomotion/patches/react-native+0.71.0.patch [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
++# LICENSE file in the root directory of this source tree.
++
++require 'net/http'
++require 'rexml/document'
++
++# This function downloads the nightly prebuilt version of Hermes based on the passed version
++# and save it in the node_module/react_native/sdks/downloads folder
++# It then returns the path to the hermes tarball
++#
++# Parameters
```
Comment (OmerGery) [inline examples/client/Locomotion/patches/react-native+0.71.0.patch:188]: remove anything thats not neccarcy in the patch files: comments, local changes , etc

File: examples/client/Locomotion/patches/react-native-config+1.5.3.patch
Comment (OmerGery) [inline examples/client/Locomotion/patches/react-native-config+1.5.3.patch:1]: remove anything thats not neccarcy in the patch files: comments, local changes , etc  - 12K Lines, probably can be less then 1000

File: examples/client/Locomotion/patches/react-native-geolocation-service+5.3.1.patch
Comment (OmerGery) [inline examples/client/Locomotion/patches/react-native-geolocation-service+5.3.1.patch:1]: remove anything thats not neccarcy in the patch files: comments, local changes , etc  - 12K Lines, probably can be less then 1000

File: examples/client/Locomotion/patches/react-native-location+2.5.0.patch
Comment (OmerGery) [inline examples/client/Locomotion/patches/react-native-location+2.5.0.patch:1]: remove anything thats not neccarcy in the patch files: comments, local changes , etc  - 12K Lines, probably can be less then 1000

File: examples/client/Locomotion/android/app/proguard-rules.pro [hunk 6]
```diff
@@ -8,3 +8,6 @@
 #   http://developer.android.com/guide/developing/tools/proguard.html
 
 # Add any project specific keep options here:
+
+-keep class com.locomotion.BuildConfig { *; }
+-keep class com.google.android.gms.location.** { *; }
```
Comment (OmerGery) [inline examples/client/Locomotion/android/app/proguard-rules.pro]: what it affects?

**Review by OmerGery — CHANGES_REQUESTED** [review]
note the local unccerrary changes

File: .idea/.gitignore
Comment (OmerGery) [inline .idea/.gitignore:1]: why its in .idea? 
just add to the regular ignore file

File: examples/client/Locomotion/android/app/proguard-rules.pro (hunk 6 above)
Comment (hilat-autofleet) [reply examples/client/Locomotion/android/app/proguard-rules.pro]: it was used to make the build work locally but now it works without it using patches

File: .idea/.gitignore
Comment (hilat-autofleet) [reply .idea/.gitignore:1]: removed

File: examples/client/Locomotion/src/Components/PageHeader/styled.js [hunk 7]
```diff
@@ -26,8 +26,8 @@ export const HeaderIconContainer = styled(Button)(({ side, theme }) => `
 `);
 
 export const HeaderIcon = styled.Image`
-  width: ${({ width }) => (width || '18px')};
-  height: ${({ height }) => (height || '18px')};
+  width: ${({ width }) => (width || 18)};
+  height: ${({ height }) => (height || 18)};
```
Comment (OmerGery) [inline examples/client/Locomotion/src/Components/PageHeader/styled.js:30]: in general check if we need these fixes to more csses

File: examples/client/Locomotion/ios/.xcode.env
Comment (OmerGery) [inline examples/client/Locomotion/ios/.xcode.env:1]: needed?

File: examples/client/Locomotion/ios/.xcode
Comment (OmerGery) [inline examples/client/Locomotion/ios/.xcode:1]: needed?

File: examples/client/Locomotion/android/build.gradle [hunk 8]
```diff
@@ -1,61 +1,24 @@
...
-        } else if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-            // For Android Users, we need to use NDK 23, otherwise the build will
-            // fail due to paths longer than the OS limit
-            ndkVersion = "23.1.7779620"
-        } else {
-            // Otherwise we default to the side-by-side NDK version from AGP.
-            ndkVersion = "21.4.7075529"
-        }
+        buildToolsVersion = '33.0.0'
+        minSdkVersion = 24
```
Comment (OmerGery) [inline examples/client/Locomotion/android/build.gradle:6]: we will check when we upload to playstore review if this minSdk is good

File: examples/client/Locomotion/App.tsx
Comment (OmerGery) [inline examples/client/Locomotion/App.tsx:1]: this is a must? i rather have minimal code changes other then the build

**Review by OmerGery — COMMENTED** [review]
Patches look good now, added some more comments

File: examples/client/Locomotion/src/Components/PageHeader/styled.js (hunk 7 above)
Comment (hilat-autofleet) [reply examples/client/Locomotion/src/Components/PageHeader/styled.js:30]: it was error while the build started, we do have more errors in the build (key is not a prop, keys are missing) 
i think treating them on another branch

File: examples/client/Locomotion/ios/.xcode.env
Comment (hilat-autofleet) [reply examples/client/Locomotion/ios/.xcode.env:1]: was added by you - thought its part of the update, i can ignore

Comment (OmerGery) [reply examples/client/Locomotion/ios/.xcode.env:1]: ah right, i think we can delete it

**Review by OmerGery — APPROVED** [review]

**Pull request merged**